code/Model/SiteTree.php

```diff
@@ -2567,6 +2567,11 @@ protected function getClassDropdown()
         foreach ($classes as $class) {
             $instance = singleton($class);
 
+            //if the current page is root and the instance can't be, exclude
+            if ($instance->config()->get('can_be_root') == false && $this->ParentID == 0) {
```

Comment:
```suggestion
            if (!$instance->config()->get('can_be_root') && $this->ParentID === 0) {
```

Reply:
I'll try that. I couldn't seem to get tests running locally for whatever reason so I've been having to leverage travis, which isn't ideal for tracking down the issue.

Reply:
This suggestion is the same functionally, just making strict comparisons where the should be 🙂

Reply:
Still no luck with local test, so debugging the failure is proving to be difficult. At a glance does it look like my methodology of testing that the class is in the source array as a key?

Reply:
Seems right - I don't have time to dive into it though. Sometimes with travis you'll have to reproduce the install steps to get the same failure. This'll mean `git clone silverstripe/silverstripe-cms`, checking out your PR branch, and then running the steps in `.travis.yml`.

Reply:
No worries, I'll give that a try and see what I can come up with.

Reply:
@ScopeyNZ that's gotten me a lot closer on local, thnx for the pointer 👏